resources/nheko.appdata.xml.in

```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2020-2022 mujx, nheko reborn developers -->
 <component type="desktop">
-  <id>@APPID@.desktop</id>
+  <id>im.nheko.Nheko.desktop</id>
```

Comment:
If we are already changing this, I think we should go with this:

```suggestion
  <id>im.nheko.Nheko</id>
```

Specifically we already changed that from io.github... to im.nheko for the flatpak. We could now stick to using nheko.desktop everywhere to not confuse normal software centers or we just change to im.nheko.Nheko everywhere for consistency. I think the latter is the way to go, even if it causes some minor complications.

See https://www.freedesktop.org/software/appstream/docs/sect-Metadata-Application.html for that reasoning.

Reply:
I'm not sure which section of it you're referring to, but I found this in that link:
>  Note that even though the component-ID can now be any rDNS name, when updating existing applications, do not change their <id/> to drop the .desktop suffix. The rules are relaxed when picking a new component-ID for new applications, but when updating older applications they still need to keep their original <id/> (when it's otherwise compliant). The ID is used to uniquely identify applications across distributions and releases and should always remain the same for the same application. 

Both will probably work relatively fine, so I'm impartial.

Reply:
For not flatpaks the id used to be: nheko.desktop
For flatpaks the id used to be until our qt6 switch: io.github.NhekoReborn.nheko

You are now changing it to im.nheko.Nheko.desktop unconditionally. Now in theory it was already that for the nightly flatpaks, but I don't think we care there. So I think the warning does not apply to us and we can drop the .desktop, since we are changing everything for the next release anyway. And there is no reason to have the .desktop anymore, I think?

Reply:
I think I agree that switching to im.nheko.Nheko is good, since I don't see a reason to prefer keeping the distro-native package id over the flatpak id.